src/stdune/list.ml

```diff
@@ -187,3 +187,6 @@ let fold_map t ~init ~f =
 
 let unzip l =
   fold_right ~init:([], []) ~f:(fun (x, y) (xs, ys) -> (x :: xs, y :: ys)) l
+
+let for_all2 x y ~f =
+  try Ok (for_all2 x y ~f) with Invalid_argument _ -> Error `Length_mismatch
```

Comment:
Can't `f` raise `Invalid_argument` though?

Reply:
I've fixed `for_all2` not to rely on exceptions.